File: 2-android-tools-and-testing/NoteKeeper/app/src/test/java/com/thedancercodes/notekeeper/DataManagerTest.java
package com.thedancercodes.notekeeper;

import org.junit.Test;

import static org.junit.Assert.*;

public class DataManagerTest {

    @Test
    public void createNewNote() throws Exception {

        // Local variable of DataManager instance
        DataManager dm = DataManager.getInstance();

        // Set up values we want in the note
        final CourseInfo course = dm.getCourse("android_async");
        final String noteTitle = "Test note title";
        final String noteText = "This is the body text of my test note";

        // Create a new note. Take the index that's returned & assign it to a local variable
        int noteIndex = dm.createNewNote();

        // Get the note associated with the index
        NoteInfo newNote = dm.getNotes().get(noteIndex);

        // Since the course is empty, set the course, title & text
        newNote.setCourse(course);
        newNote.setTitle(noteTitle);
        newNote.setText(noteText);

        // Go back to the DataManager and get a note at the defined index
        NoteInfo compareNote = dm.getNotes().get(noteIndex);

        // GOAL: Make sure compareNote contains the things that we have put into the note
        // at that spot.
        assertEquals(course, compareNote.getCourse());
        assertEquals(noteTitle, compareNote.getTitle());
        assertEquals(noteText, compareNote.getText());

    }
}

File: Kotlin/NoteKeeper/2-tools-and-testing/NoteKeeper/app/src/test/java/com/thedancercodes/notekeeper/DataManagerTest.kt
package com.thedancercodes.notekeeper

import org.junit.Test

import org.junit.Assert.*

class DataManagerTest {

    @Test
    fun addNote() {

        // Variables to create a new note
        val course = DataManager.courses.get("android_async")!!
        val noteTitle = "This is test note"
        val noteText = "This is the body of my text note"

        // Index value returned when we create a new note
        val index = DataManager.addNote(course, noteTitle, noteText)

        // Get the note at that index
        val note = DataManager.notes[index]

        // Check that the note we get back has the values in it that we expect
        assertEquals(course, note.course)
        assertEquals(noteTitle, note.title)
        assertEquals(noteText, note.text)

    }

    @Test
    fun findSimilarNotes() {
        val course = DataManager.courses.get("android_async")!!
        val noteTitle = "This is test note"
        val noteText1 = "This is the body of my test note"
        val noteText2 = "This is the body of my second test note"

        // Add 2 notes
        val index1 = DataManager.addNote(course, noteTitle, noteText1)
        val index2 = DataManager.addNote(course, noteTitle, noteText2)

        val note1 = DataManager.findNote(course, noteTitle, noteText1)
        val foundIndex1 = DataManager.notes.indexOf(note1)
        assertEquals(index1, foundIndex1)

        val note2 = DataManager.findNote(course, noteTitle, noteText2)
        val foundIndex2 = DataManager.notes.indexOf(note2)
        assertEquals(index2, foundIndex2)




    }
}